File: pkg/model/credential/credential.go
package credential

import (
	"github.com/KubeOperator/KubeOperator/pkg/model/common"
	uuid "github.com/satori/go.uuid"
)

type Credential struct {
	common.BaseModel
	ID         string
	Name       string `gorm:"not null;unique"`
	Username   string
	Password   string
	PrivateKey string
	Type       string
}

func (c *Credential) BeforeCreate() (err error) {
	c.ID = uuid.NewV4().String()
	return err
}

func (c Credential) TableName() string {
	return "ko_credential"
}

File: ui/src/app/business/setting/credential/credential.ts
import {BaseModel, BaseRequest} from '../../../shared/class/BaseModel';

export class Credential extends BaseModel {
    id: string;
    name: string;
    username: string;
    password: string;
    privateKey: string;
    type = 'password';
}


export class CredentialCreateRequest extends BaseRequest {
    id: string;
    name: string;
    username: string;
    password: string;
    privateKey: string;
    type: string;
}
